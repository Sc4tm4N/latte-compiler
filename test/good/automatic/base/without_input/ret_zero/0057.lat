// parameters override for base types

void b(boolean input) {
  input = false;
  if(!input)
    printString("b2 OK");
}

void i(int input) {
  input = 33;
  if(input == 33)
    printString("i2 OK");
}

void s(string input) {
  input = "kappa";
  if(input == "kappa")
    printString("s2 OK");
}

int main() {
  boolean b = true;
  if(b)
    printString("b1 OK");
  b(b);
  if(b)
    printString("b3 OK");

  int i = 13;
  if(i == 13)
    printString("i1 OK");
  i(i);
  if(i == 13)
    printString("i3 OK");

  string s = "delta";
  if(s == "delta")
    printString("s1 OK");
  s(s);
  if(s == "delta")
    printString("s3 OK");

  return 0;
}
