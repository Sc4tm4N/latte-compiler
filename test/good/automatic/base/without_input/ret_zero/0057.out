b1 OK
b2 OK
b3 OK
i1 OK
i2 OK
i3 OK
s1 OK
s2 OK
s3 OK
